tools/toolchains/iar.py

```diff
@@ -54,7 +60,10 @@ def __init__(self, target, notify=None, macros=None, build_profile=None,
             build_profile=build_profile
         )
 
-        self.check_c_lib_supported(target, "iar")
+        # Build with the standard C lib if c_lib is set to small as there
+        # is no small lib for the IAR toolchain
+        if target.c_lib.lower() != "small":
```

Comment:
If `c_lib` is set to "small" for IAR, we want to check that "std" is supported and if so revert to using standard.

Reply:
[Done](https://github.com/ARMmbed/mbed-os/compare/f4d874d12274bf7090233e5dc9bec5705ee09994..bc8aeb6cd2ff42de8f75808940ea81db385827b2).